src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -1125,6 +1125,63 @@ protected static void VerifyOperationTreeForTest<TSyntaxNode>(CSharpCompilation
             additionalOperationTreeVerifier?.Invoke(actualOperation, compilation, syntaxNode);
         }
 
+        protected static void VerifyFlowGraphForTest<TSyntaxNode>(CSharpCompilation compilation, string expectedFlowGraph)
+            where TSyntaxNode : SyntaxNode
+        {
+            var tree = compilation.SyntaxTrees[0];
+            var model = compilation.GetSemanticModel(tree);
+            SyntaxNode syntaxNode = GetSyntaxNodeOfTypeForBinding<TSyntaxNode>(GetSyntaxNodeList(tree));
+
+            ImmutableArray<Operations.BasicBlock> graph = model.GetControlFlowGraph((Operations.IBlockOperation)model.GetOperation(syntaxNode));
+            var map = new Dictionary<Operations.BasicBlock, int>();
+
+            for (int i = 0; i < graph.Length; i++)
+            {
+                map.Add(graph[i], i);
+            }
+
+            var stringBuilder = PooledObjects.PooledStringBuilder.GetInstance();
+
+            for (int i = 0; i < graph.Length; i++)
+            {
+                var block = graph[i];
+                stringBuilder.Builder.AppendLine($"Block[{i}] - {block.Kind}");
+
+                var predecessors = block.Predecessors;
+
+                if (!predecessors.IsEmpty)
+                {
+                    stringBuilder.Builder.AppendLine($"    Predecessors ({predecessors.Count})");
```

Comment:
Nit: consider printing the list of predecessors on the same line for compactness. Then maybe the count isn't so useful, since typically 1 or 2 and clearly visible.